test/NuGet.Core.Tests/NuGet.Protocol.Core.v3.Tests/HttpSource/HttpRetryHandlerTests.cs

```diff
@@ -141,13 +141,18 @@ public class HttpRetryHandlerTests
                 return new HttpResponseMessage(HttpStatusCode.InternalServerError);
             };
 
+            // Make a timeout that is greater than SmallTimeout (the duration of the request, in 
+            // this case) but less than the total time spent on all attempts, which is MaxTries
+            // times SmallTimeout.
+            var requestTimeout = new TimeSpan((SmallTimeout.Ticks * MaxTries) / 2);
```

Comment:
What would happen when all `MaxTries` timed out? I think `HttpRetryHandler` would re-throw the timeout exception and that fails the test because `SendAsync` below doesn't catch it.